scripts/migrations/2020-10-archivematica_bags_migration.py

```diff
@@ -0,0 +1,246 @@
+import sys
+import os
+import shutil
+import hashlib
+from tqdm import tqdm
+
+from common import (
+    get_aws_resource,
+    get_aws_client,
+    get_storage_client,
+    get_secret,
+    get_elastic_client,
+)
+
+
+def query_index(elastic_client, elastic_index, elastic_query, transform):
+    initial_query = elastic_client.search(
+        index=elastic_index, body=elastic_query, size=0
+    )
+
+    document_count = initial_query["hits"]["total"]["value"]
+    results = helpers.scan(
+        client=elastic_client, index=elastic_index, size=5, query=elastic_query
+    )
+
+    with tqdm(total=document_count, file=sys.stdout) as pbar:
+        for result in results:
+            transform(result)
+            sys.exit(1)
+            pbar.update(1)
+
+
+def bag_creator(
+    workflow_s3_client, storage_s3_client, storage_client, s3_upload_bucket
+):
+    def create_bag(result):
+        document = result["_source"]
+
+        id = document["id"]
+        bucket = document["location"]["bucket"]
+        path = document["location"]["path"]
+        version = f"v{document['version']}"
+        space = document["space"]
+        external_identifier = document["info"]["externalIdentifier"]
+
+        provider = document["location"]["provider"]
+        assert provider == "amazon-s3"
+
+        path_prefix = f"s3://{bucket}/{path}"
+
+        working_id = id.replace("/", "_")
+        target_folder = "target"
+        working_folder = f"{target_folder}/{working_id}"
+        tagmanifest_name = "tagmanifest-sha256.txt"
+        archive_name = f"{working_id}.tar.gz"
+        s3_upload_prefix = "born-digital/archivematica-uuid-update"
+
+        if not os.path.exists(working_folder):
+            os.makedirs(working_folder)
+
+        prefix_patterns = ["bagit.txt", "bag-info.txt", "manifest-", "tagmanifest-"]
+
+        def _write_fetch_file():
+            def _create_fetch_line(file):
+                return f"{path_prefix}/{file['path']}\t{file['size']}\t{file['name']}\n"
+
+            files = document["files"]
+
+            fetch_file = open(f"{working_folder}/fetch.txt", "w")
+            for file in files:
+                fetch_file.write(_create_fetch_line(file))
+            fetch_file.close()
+
+        def _filter_bag_files(prefix):
+            response = storage_s3_client.list_objects_v2(
+                Bucket=bucket, Prefix=f"{path}/{version}/{prefix}"
+            )
+
+            if "Contents" in response:
+                return [content["Key"] for content in response["Contents"]]
+            else:
+                return []
+
+        def _check_expected_prefixes_exist(file_locations):
+            for prefix in prefix_patterns:
+                found_match = False
+
+                for location in file_locations:
+                    if prefix in location:
+                        found_match = True
+
+                if not found_match:
+                    raise RuntimeError(f"Missing any files matching prefix: {prefix}")
```

Comment:
Python has a convenience method you might find helpful

https://docs.python.org/3/library/functions.html?highlight=any#any:

> Return True if any element of the iterable is true. If the iterable is empty, return False.

Consider:

```suggestion
            for prefix in prefix_patterns:
                if not any(prefix in location for location in file_locations):
                    raise RuntimeError(f"Missing any files matching prefix: {prefix}")
```